### PR TITLE
Add support for insights-client legacy APIs forwarding

### PR DESCRIPTION
**Pull request by ShimShtein**

### CODE REVIEW

File: lib/foreman_rh_cloud.rb [hunk 1]
```diff
@@ -12,6 +12,10 @@ def self.cert_base_url
     @cert_base_url ||= ENV['SATELLITE_CERT_RH_CLOUD_URL'] || 'https://cert.cloud.redhat.com'
   end
 
+  def self.legacy_inights_url
+    @legacy_inights_url ||= ENV['SATELLITE_LEGACY_INSIGHTS_URL'] || 'https://cert-api.access.redhat.com'
```
Comment (lphiri) [inline lib/foreman_rh_cloud.rb:16]: Nit - I assume you want `@legacy_insights_url`?

**Review by lphiri — APPROVED** [review]
Minor naming nit, but otherwise looks ok.

**Comment by ShimShtein** [comment]
@laviro, ready for review

**Comment by ShimShtein** [comment]
Thanks @lphiri, fixed the name

File: lib/foreman_rh_cloud.rb [hunk 2]
```diff
@@ -12,6 +12,10 @@ def self.cert_base_url
     @cert_base_url ||= ENV['SATELLITE_CERT_RH_CLOUD_URL'] || 'https://cert.cloud.redhat.com'
   end
 
+  def self.legacy_inights_url
```
Comment (laviro) [inline lib/foreman_rh_cloud.rb:15]:
```suggestion
  def self.legacy_insights_url
```

File: app/services/foreman_rh_cloud/cloud_request_forwarder.rb [hunk 3]
```diff
@@ -76,6 +58,36 @@ def prepare_forward_params(original_request, branch_id)
...
+        }
+      when platform_request?
+        {
+          url: ForemanRhCloud.cert_base_url + request_path.sub('/redhat_access/r/insights/platform', '/api'),
+          ssl_client_cert: OpenSSL::X509::Certificate.new(certs[:cert]),
+          ssl_client_key: OpenSSL::PKey::RSA.new(certs[:key]),
+        }
+      else # Legacy insights API
+        {
+          url: ForemanRhCloud.legacy_inights_url + request_path.sub('/redhat_access/r/insights', '/r/insights'),
```
Comment (laviro) [inline app/services/foreman_rh_cloud/cloud_request_forwarder.rb:75]:
```suggestion
          url: ForemanRhCloud.legacy_insights_url + request_path.sub('/redhat_access/r/insights', '/r/insights'),
```

**Review by laviro — COMMENTED** [review]
Just the small nitpick, otherwise looks great!

**Comment by ShimShtein** [comment]
@laviro, done!

**Review by laviro — APPROVED** [review]
Looks good to me, thanks @ShimShtein !

**Pull request merged**